io_scene_nif/modules/nif_export/geometry/mesh/__init__.py

```diff
@@ -108,24 +108,11 @@ def export_tri_shapes(self, b_obj, n_parent, trishape_name=None):
             mesh_materials = [None]
 
         # is mesh double sided?
-        mesh_doublesided = b_mesh.show_double_sided
+        # todo [mesh/material] detect from material settings!
+        mesh_doublesided = True
 
         # vertex color check
-        mesh_hasvcol = False
-        mesh_hasvcola = False
-
-        if b_mesh.vertex_colors:
-            mesh_hasvcol = True
-
-            # vertex alpha check
-            if len(b_mesh.vertex_colors) == 1:
-                NifLog.warn("Mesh only has one Vertex Color layer. Default alpha values will be written."
-                            "For Custom alpha values add a second vertex layer, greyscale only")
-            else:
-                for b_loop in b_mesh.vertex_colors[1].data:
-                    if b_loop.color.v > NifOp.props.epsilon:
-                        mesh_hasvcola = True
-                        break
+        mesh_hasvcol = b_mesh.vertex_colors
```

Comment:
Does this support alpha

Reply:
yeah it does, we now have rgba alpha layers. I gotta rewrite the alpha property detection tho

Reply:
I am not sure if the tooling supports it though.

Reply:
I haven't kept up to date, but last time I checked it was workable albeit very crude. Vcols seem to be getting a rewrite anyway for 2.9 lol